src/server/tcp.rs

```diff
@@ -51,6 +51,19 @@ impl AddrIncoming {
     pub fn bind(addr: &SocketAddr) -> crate::Result<Self> {
         AddrIncoming::new(addr)
     }
+    
+    /// Creates a new `AddrIncoming` from an existed `TcpListener`
+    pub fn from_listener(listener: TcpListener) -> crate::Result<Self> {
```

Comment:
With this addition, `from_std` can be written to call it directly, ya?

Reply:
Yes, of course.